pvlib/pvsystem.py

```diff
@@ -1172,6 +1164,51 @@ def calcparams_desoto(effective_irradiance, temp_cell,
     return IL, I0, Rs, Rsh, nNsVth
 
 
+def sdm_campanelli(F, H, **kwargs):
+    """
+    Compute the Campanelli et al. model
+
+    TODO Details
+    """
+
+    return singlediode(*get_coeffs_campanelli(F, H, **kwargs))
+
+
+def get_coeffs_campanelli(F, H, **kwargs):
```

Comment:
Why use ``**kwargs`` here? Would we not want to use positional arguments for everything that's required and does not have a default, and use named keyword arguments for everything that has a default?

Reply:
Unless someone can show that this is less performant, I think it is less fragile and thus preferable to `_build_kwargs`. Of course, the required `kwargs` must be well-documented in the "base" function. Given the considerable level of indirect inference and "kwargs packing" that occurs throughout the current architecture, I don't find this any more obscure than the status quo. I realize some might disagree here and prefer positional arguments, but I personally really prefer removing order dependence whenever possible in function arguments (esp. for model parameters like `IL` as opposed to control variables like `effective_irradiance`). On a related note, Python3 has [keyword-only arguments](https://www.python.org/dev/peps/pep-3102/) that don't have to specify a default value :).

Reply:
Python makes no distinction. The user can provide keyword arguments or positional arguments whenever they want unless the function signature contains splat operators like `*a, **kw`, which are only used to match inherited signatures, or to collect unknown user supplied extra arguments.

`import this` says explicit is better than implicit. IMHO unless you are collecting args for inherited signatures or slurring the user to supply their own keyword arguments, this usage should be avoided, and I believe most linters would warn

This

Reply:
K. So I am going to look up if there is a Python2 backport of keyword-only arguments that don't require a default value. I think this would be a fair compromise to my preference to avoid [the complexity of order](https://www.youtube.com/watch?v=rI8tNMsozo0) while also being uber-explicit in the low-level functions.

The only downside that I see here is potentially long keyword names, but again I think explicitness and "self-documenting" function readability trumps lengthiness, and think my names are short but sufficiently expressive and will of course be fully documented.

Reply:
I realize that @thunderfish24 and others will think this a step backwards, but I'd prefer `calcparams_campanelli` to `get_coeffs_campanelli` to keep the function name pattern parallel to the other version of single diode models. When we want to change this API pattern we can do it for all single diode model coefficient functions.

Reply:
No prob. Consistency is always an important consideration.

Reply:
I agree with @cwhanse that ``calcparams_campanelli`` is preferable here. My intention (perhaps poorly communicated elsewhere) is for all ``get_*`` functions/methods to be wrappers for other functions/methods. See #427 for more on that pattern. This PR is big enough already, so let's not add a new wrapper here. 

Re keyword-only arguments... I am not aware of a way to do that in Python 2 without some miserable hacks. Users can still call positional arguments as keyword arguments in any order you'd like, though I think it's good to be consistent within pvlib itself.

Reply:
I agree that Python2 hacks look really bad, so I'll probably have to give up and simply used named positional arguments :(.